library/classes/Document.class.php

```diff
@@ -695,6 +875,87 @@ function createDocument(
         return '';
     }
 
+    /**
+     * Retrieves the document data that has been saved to the filesystem or couch db.  If the $force_no_decrypt flag is
+     * set to true, it will return the encrypted version of the data for the document.
+     * @param bool $force_no_decrypt True if the document should have its data returned encrypted, false otherwise
+     * @throws BadMethodCallException Thrown if the method is called when the document has been marked as deleted or expired
+     * @return false|string Returns false if the data failed to decrypt, or a string if the data decrypts or is unencrypted.
+     */
+    function get_data($force_no_decrypt = false)
+    {
+        $storagemethod = $this->get_storagemethod();
+
+        if ($this->has_expired()) {
+            throw new BadMethodCallException("Should not attempt to retrieve data from expired documents");
+        }
+        if ($this->is_deleted()) {
+            throw new BadMethodCallException("Should not attempt to retrieve data from deleted documents");
+        }
+
+        $base64Decode = false;
+
+        if ($storagemethod === self::STORAGE_METHOD_COUCHDB) {
+            $base64Decode = true;
```

Comment:
actually, a clearer way to do what i posted below may be to instead do the following here:
```
if ($this->is_encrypted()) {
    $base64Decode = false;
} else {
    $base64Decode = true;
}
```

Reply:
This should now be fixed.